spec/support/controller_macros.rb

```diff
@@ -0,0 +1,9 @@
+module ControllerMacros
+	
+  def set_devise_mapping_to_user
+		before(:each) do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+    end
+	end
+  
+end
```

Comment:
[as per the Devise documentation](https://github.com/heartcombo/devise/wiki/How-To:-Test-controllers-with-Rails-(and-RSpec)#controller-specs).